docs/docs/recipes.md

```diff
@@ -241,3 +241,26 @@ You can use constraints to restrict where to run the tasks for your apps. See
 The advantage of using constraints to restrict where tasks run is that you only have to provide appropriate attributes on the Mesos agents.
 
 The disadvantages are that nothing prevents another framework from scheduling tasks on these agents and consuming all the resources your special tasks need. To avoid this, you need to configure Marathon so that all non-special apps are constrained *not* to use the special nodes.
+
+## Running a Single Instance Application
+
+Often legacy applications are built in a way which prevents running multiple instances of that application concurrently. Marathon supports a label which allows you to prevent the user scaling an application beyond a single instance.
+
+In your application definition, set the label `MARATHON_SINGLE_INSTANCE_APP` to true:
+
+```json
+  "labels":{
+    "MARATHON_SINGLE_INSTANCE_APP": "true",
+  }
+```
+
+It's also necessary to specify an `upgradeStrategy` - this permits Marathon to kill all instances of this application when upgrading (otherwise it would be impossible to upgrade), and prevents Marathon from launching more than one instance:
```

Comment:
I'd get rid of the dash. It's not only necessary to specify an upgradeStrategy; it's necessary to specify _an upgradeStrategy that permits Marathon to kill all instances_. Apps with label `SINGLE_INSTANCE_APP` will fail validation (and thus any deployment) if their upgradeStrategy is unequal to `UpgradeStrategy(minimumHealthCapacity = 0, maximumOverCapacity = 0))`.

Moreover, the part _otherwise it would be impossible to upgrade_ is not correct. In general, Marathon will spawn new instances if it cannot start the upgrade by killing at least one task – it handles the upgradeStrategy as a proposal, not as a command.